arctic/store/_ndarray_store.py

```diff
@@ -268,14 +293,21 @@ def _do_read(self, collection, version, symbol, index_range=None):
         segment_count = None
 
         spec = {'symbol': symbol,
-                'parent': version_base_or_id(version),
                 'segment': {'$lt': to_index}
                 }
         if from_index is not None:
             spec['segment']['$gte'] = from_index
         else:
             segment_count = version.get('segment_count', None)
 
+        # We want to use FW pointers to read data if:
+        #  ARCTIC_FORWARD_POINTERS is currently enabled/hybrid AND last version was written with FW pointers,
+        #  otherwise fall back to original implementation
+        if ARCTIC_FORWARD_POINTERS is not FwPointersCfg.DISABLED and FW_POINTERS_KEY in version:
```

Comment:
I think this `if` should always read via fwd pointers if the field is there, as:
- intent is to move to this anyway
- if someone doesn’t set the environment variable, then reads to fwd pointer only libs will fail. 

Reads should be forwards compatible in new versions of arctic. Users shouldn’t need to set variables to read data.

Reply:
plase check the new implementation: _spec_fw_pointers_aware()
we no longer rely on env/global variables.

I had already this change in flight, it seems unfortunate that you took the time to review the outdated code.

Reply:
I am resolving this now, the new implementation if code detects fw-pointers info on the version document, it is just using this, without dependency on the environment vars, except FW are explicitly disabled by user.